client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -716,31 +716,41 @@ Undocumented.prototype.validateDomainContactInformation = function (
 /**
  * Validates the specified Google Apps contact information
  *
+ * The contactInformation keys can be in camelCase or snake_case, and will be
+ * converted to snake_case before being submitted. The returned data keys will
+ * be converted to camelCase before being passed to the callback or the resolved
+ * Promise.
+ *
  * @param {object} contactInformation - user's contact information
- * @param {Function} callback The callback function
- * @returns {Promise} A promise that resolves when the request completes
+ * @param {string[]} domainNames - domain names for which GSuite is being purchased
+ * @param {(error: string, data: object) => void} [callback] The callback function.
+ * @returns {Promise|undefined} If no callback, returns a Promise that resolves when the request completes
  */
 Undocumented.prototype.validateGoogleAppsContactInformation = function (
 	contactInformation,
+	domainNames,
 	callback
 ) {
-	const data = mapKeysRecursively( { contactInformation }, snakeCase );
+	const { contact_information } = mapKeysRecursively( { contactInformation }, snakeCase );
+	debug( '/me/google-apps/validate', contact_information, domainNames );
 
-	return this.wpcom.req.post(
-		{ path: '/me/google-apps/validate' },
-		data,
-		( error, successData ) => {
-			if ( error ) {
-				return callback( error );
-			}
+	const stripHeadersFromHttpResponse = ( httpResponse ) =>
+		Object.keys( httpResponse )
+			.filter( ( key ) => key !== '_headers' )
+			.reduce( ( newResponse, key ) => ( { ...newResponse, [ key ]: httpResponse[ key ] } ), {} );
```

Comment:
This is a good question. My instinct is that yes, we are moving away from lodash because vanilla JS has finally caught up with most of the things that lodash was made to do, and because that's one less library in our bundle. But as I was trying to justify my response I realized it may just be biased toward removing something that I see as added complexity. 

This sparked a good conversion in Slack, which seems to be pretty evenly split around this topic. On the one hand, `omit` here would be much easier to read, but under the hood it's a _massive_ library that uses its own functions quite a lot so it's pulling in hundreds of lines of code versus these three. 

The ideal solution is probably to take these three lines and make our own `omit` function, which is the best of both worlds; efficient with code and concise.